samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2420,206 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
+            )
+            func_config.MaxBatchSize = passthrough_value(function.MaxBatchSize)
+            func_config.Description = passthrough_value(function.Description)
+            func_config.Runtime = self._parse_runtime(function, resolver_code_settings, relative_id)
+
+            if function.Sync:
+                func_config.SyncConfig = self._parse_function_sync(function.Sync)
+
+            resources.append(func_config)
+
+        return resources
+
+    def _check_and_construct_none_datasource(
+        self,
+        functions: Dict[str, aws_serverless_graphqlapi.Function],
+        api_id: Intrinsicable[str],
+        none_datasource_logical_id: str,
+    ) -> Optional[DataSource]:
+        """
+        Checks if a DataSource with type "NONE" needs to be created.
+
+        Within a Serverless::GraphQLApi Function resource, customers can create a
+        DataSource of Type "NONE" for quick use in Functions. To do so, a customer
+        can input "NONE", "None", or "none" in the "DataSource" property. Only one
+        DataSource will be created for each GraphQLApi, and all GraphQLApi functions
+        with this none input will reference it.
+        """
+        for function in functions.values():
+            if not function.DataSource or function.DataSource not in NONE_DATASOURCE_LITERALS:
+                continue
+
+            none_datasource = DataSource(
+                logical_id=none_datasource_logical_id, depends_on=self.depends_on, attributes=self.resource_attributes
+            )
+            none_datasource.ApiId = api_id
+            none_datasource.Name = none_datasource_logical_id
+            none_datasource.Type = "NONE"
+
+            return none_datasource
+
+        return None
+
+    @staticmethod
+    def _parse_datasource_name(
+        relative_id: str,
+        function: aws_serverless_graphqlapi.Function,
+        datasources: Optional[Dict[str, aws_serverless_graphqlapi.DynamoDBDataSource]],
+        none_datasource_logical_id: str,
+    ) -> str:
+        """
+        Parse DataSource name from a Serverless::GraphQLApi function.
+
+        There are 3 different cases for the DataSource name:
+
+        1. Customer defines the "DataSourceName" property, and we can simply return this value.
+
+        2. Customer defines "DataSource" propery as "NONE", so we return the name of the NoneDataSource
+           for this Serverless::GraphQLApi resource.
+
+        3. Customer defines "DataSource" property with a logical id of a datasource defined in
+           Serverless::GraphQLApi. We can then search if this DataSource exists, and return the name.
+           If it does not exist, throw an InvalidResourceException.
+        """
+        if function.DataSourceName:
+            return function.DataSourceName
+
+        if function.DataSource in NONE_DATASOURCE_LITERALS:
+            return none_datasource_logical_id
+
+        if not datasources or function.DataSource not in datasources:
```

Comment:
no need in `not datasources`

Reply:
`datasources` is an optional value. It will throw a `TypeError` if it's None so I have to do a None check first.